.github/workflows/lint.yml

```diff
@@ -0,0 +1,21 @@
+name: golangci-lint
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
Ooops sorry commented in the wrong spot on your fork: Can you run on PR against main as well? I think we want to know before we merge to main if linting is ok right?